src/axom/inlet/SchemaCreator.hpp

```diff
@@ -130,8 +130,162 @@ class SchemaCreator
    * \return Shared pointer to the created Field
    *****************************************************************************
    */
-  std::shared_ptr<Field> addString(const std::string& name,
-                                   const std::string& description);
+  virtual std::shared_ptr<Field> addString(const std::string& name,
+                                   const std::string& description) = 0;
+  /*!
+   *****************************************************************************
+   * \brief Return whether a Table with the given name is present in this Table's subtree.
+   *
+   * \return Boolean value indicating whether this Table's subtree contains this Table.
+   *****************************************************************************
+   */
+  virtual bool hasTable(const std::string& tableName) = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Return whether a Field with the given name is present in this Table's
+   *  subtree.
+   *
+   * \return Boolean value indicating whether this Table's subtree contains this Field.
+   *****************************************************************************
+   */
+  virtual bool hasField(const std::string& fieldName) = 0;
+
+  /*!
+   *****************************************************************************
+   * \return An unordered map from Field names to the child Field pointers for 
+   * this Table.
+   *****************************************************************************
+   */
+  virtual std::unordered_map<std::string, std::shared_ptr<Field>> getChildFields() = 0;
```

Comment:
Should this be returned as a `const` reference or by reference to avoid copying the map?

Reply:
This is probably true.  I need to do a general once over of what is passed around for efficiency.